app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -22,142 +22,163 @@
+            findEditSetRecord(editSetId, recordId).flatMap { recordOutcome =>
+              logger.info(s"The edit set id is $editSetId for record id $recordId")
+              recordOutcome.flatMap(getSubmitAction) match {
+                case Right(Save(editSetRecord, values)) => submit(editSetId, editSetRecord, values)
+                case Right(Discard) => Redirect(controllers.routes.EditSetRecordController.discard(editSetId, recordId))
+                case Right(CalculateDates(record))    => calculateDates(user, editSet, record)
+                case Right(AddAnotherCreator(record)) => addAnotherCreator(user, editSet, record)
+                case Right(RemoveLastCreator(record)) => removeLastCreator(user, editSet, record)
+                case Left(FormValidationFailed(formWithErrors, record)) =>
+                  BadRequest(generateEditSetRecordEditView(user, editSet, record, formWithErrors))
+                case Left(EditSetNotFound(missingEditSetId)) =>
+                  BadRequest(s"Edit Set with ID [$missingEditSetId] not found")
+                case Left(EditSetRecordNotFound(missingOci)) => BadRequest(s"Record with $missingOci not found")
+                case Left(InvalidAction(badAction))          => BadRequest(s"$badAction is not allowed action")
+                case Left(MissingAction)                     => BadRequest("This action is not allowed")
+              }
+            }
+          case Left(_) => BadRequest(s"Edit Set [$editSetId] was not found")
         }
       }
-    }
   }
 
-  def save(id: String, oci: String): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
+  private def submit(
+    editSetId: String,
+    editSetRecord: EditSetRecord,
+    editSetRecordFormValues: EditSetRecordFormValues
+  ): Future[Result] =
+    editRecordSetService
+      .updateEditSetRecord(editSetId, editSetRecord.oci, editSetRecordFormValues)
+      .unsafeToFuture()
+      .map { updateResponseStatus =>
+        logger
+          .info(s"We are currently ignoring the returned status of the update: [$updateResponseStatus]")
+        Redirect(controllers.routes.EditSetRecordController.save(editSetId, editSetRecord.oci))
+      }
+
+  def save(editSetId: String, oci: String): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
     withUserAsync { user =>
-      findRecord(id, oci).map {
-        case Right(record) =>
-          val title = resolvedMessage(MessageKeys.title)
-          val editSetName = editSets.getEditSet().name
-          val heading = resolvedMessage(MessageKeys.heading, record.ccr)
-          val message = resolvedMessage(MessageKeys.buttonSave)
-          val recordType = record.recordType
-          logger.info(s"Save changes for record id $oci edit set id $id")
-          Ok(editSetRecordEditSave(user, editSetName, title, heading, oci, message, recordType))
-
-        case Left(_) => NotFound
+      findEditSet(editSetId).flatMap {
```

Comment:
I think we could use a for comprension here rather than `flatMap` followed by `map`

Reply:
I'll try that.

Reply:
Changed.

Reply:
Resolving as @felixgb approved the PR.